tests/functional/test_change_password.py

```diff
@@ -0,0 +1,91 @@
+import os
+import jwt
+from tests.factories import user
+
+
+class TestChangePassword(object):
+    """A User can change password"""
+
+    def test_change_password_with_correct_credentials(self, client, current_user):
+        """Post to /change-password url with correct credentials succeeds"""
+
+        res = client.post(
+            "/change-password",
+            json={
+                "email": current_user.email,
+                "old_password": "123123",
+                "new_password": "abc123",
+            },
+        )
+
+        assert res.status_code == 204
```

Comment:
you should also assert that the side effect that you want (the password changes) has occurred.  I believe there's a way to do this with a context manager.